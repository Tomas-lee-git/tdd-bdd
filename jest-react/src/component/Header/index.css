.header {
  height: 60px;
  background-color: #333;
  font-size: 24px;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
}
.header-container {
  width: 500px;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.header-input {
  width: 400px;
  margin-left: 50px;
  border-radius: 5px;
  height: 24px;
  outline: none;
}
