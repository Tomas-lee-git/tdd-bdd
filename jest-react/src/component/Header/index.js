import React, { Component } from 'react';
import './index.css';
class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    };
  }
  handleChange = e => {
    const { value } = e.target;
    this.setState({
      value: value,
    });
  };

  handleKeyUp = e => {
    const { value } = this.state;
    const { addUndoItem } = this.props;
    if (e.keyCode === 13 && value) {
      addUndoItem(value);
      this.setState({
        value: '',
      });
    }
  };

  render() {
    const { value } = this.state;
    return (
      <div className="header">
        <div className="header-container">
          <span>TodoList</span>
          <input
            className="header-input"
            type="text"
            data-test="inputItem"
            value={value}
            placeholder="addTodo"
            onChange={this.handleChange}
            onKeyUp={this.handleKeyUp}
          />
        </div>
      </div>
    );
  }
}

export default Header;
