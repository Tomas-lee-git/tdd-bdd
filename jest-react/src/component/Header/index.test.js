import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Header from './index';

Enzyme.configure({ adapter: new Adapter() });
describe('Header component test', () => {
  let wrapper = null;
  let inputItem = null;
  const userInput = 'TDD实战';

  beforeEach(() => {
    wrapper = shallow(<Header />);
    inputItem = wrapper.find("[data-test='inputItem']");
  });

  describe('dom test', () => {
    test('should have one input ', () => {
      expect(inputItem.length).toEqual(1);
    });
  });

  describe('input value test', () => {
    test('input value should be "" ', () => {
      expect(inputItem.prop('value')).toEqual('');
    });

    test('input 的值应该和用户输入的值保持一致', () => {
      inputItem.simulate('change', {
        target: {
          value: userInput,
        },
      });
      expect(wrapper.state('value')).toEqual(userInput);

      // dom 变化之后，需要重新获取dom，才能获得相应的变动
      // inputItem = wrapper.find("[data-test='inputItem']");
      // expect(inputItem.prop('value')).toEqual(userInput);
    });
  });

  describe('按下"Enter"键，给list添加数据的方法是否应该被调用', () => {
    let fn = null;
    beforeEach(() => {
      fn = jest.fn();
      wrapper = shallow(<Header addUndoItem={fn} />);
      inputItem = wrapper.find("[data-test='inputItem']");
    });

    test('如果input没有内容，不应该被调用”', () => {
      wrapper.setState({ value: '' });
      inputItem.simulate('keyUp', {
        keyCode: 13,
      });
      expect(fn).toHaveBeenCalledTimes(0);
    });

    test('如果input有内容，应该被调用', () => {
      wrapper.setState({ value: userInput });
      inputItem.simulate('keyUp', {
        keyCode: 13,
      });
      console.log(wrapper.state('value'));
      expect(fn).toHaveBeenCalledTimes(1);
      expect(fn).toHaveBeenCalledWith(userInput);
    });
  });
});
