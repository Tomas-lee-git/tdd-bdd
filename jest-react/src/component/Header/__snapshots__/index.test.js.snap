// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test dom test snapshot test 1`] = `
<div
  className="header"
>
  <div
    className="header-container"
  >
    <span>
      TodoList
    </span>
    <input
      className="header-input"
      data-test="inputItem"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="addTodo"
      type="text"
      value=""
    />
  </div>
</div>
`;
